Firestore/core/src/firebase/firestore/local/serializer.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_LOCAL_SERIALIZER_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_LOCAL_SERIALIZER_H_
+
+#include "Firestore/core/src/firebase/firestore/model/maybe_document.h"
+#include "Firestore/core/src/firebase/firestore/nanopb/reader.h"
+#include "Firestore/core/src/firebase/firestore/nanopb/writer.h"
+#include "Firestore/core/src/firebase/firestore/remote/serializer.h"
+#include "Firestore/core/src/firebase/firestore/util/status.h"
+
+namespace firebase {
+namespace firestore {
+namespace local {
+
+/**
+ * @brief Serializer for values stored in the LocalStore
+ *
+ * Note tha local::Serializer currently delegates to the remote::Serializer (for
+ * the Firestore v1beta1 RPC protocol) to save implementation time and code
+ * duplication. We'll need to revisit this when the RPC protocol we use diverges
+ * from local storage.
+ */
+class Serializer {
+ public:
+  Serializer(const remote::Serializer& rpc_serializer)
```

Comment:
Explicit? If this is deliberate, please add a brief comment stating that.

Reply:
Done.